include/ginkgo/core/matrix/dense.hpp

```diff
@@ -1163,6 +1186,14 @@ class Dense
      */
     virtual void compute_norm1_impl(LinOp* result) const;
 
+    /**
+     * @copydoc compute_squared_norm2(LinOp*) const
+     *
+     * @deprecated  This function will be removed in the future,
+     *              we will instead always use Ginkgo's implementation.
+     */
+    virtual void compute_squared_norm2_impl(LinOp* result) const;
```

Comment:
I see. if you need to provide this for override, I will change my comment above. you can do compute_squred_norm(LinOp*) -> compute_squred_norm_impl(LinOp*) // create tmp-> compute_squred_norm(LinOp*, tmp)